support/refc/runtime.c

```diff
@@ -112,44 +113,41 @@ Value *tailcall_apply_closure(Value *_clos, Value *arg) {
   return (Value *)makeClosureFromArglist(f, newArgs);
 }
 
-int extractInt(Value *v) {
+int idris2_extractInt(Value *v) {
```

Comment:
I think ideally it would either be safe to use `idris2_extractInt` any time an integer value was in play or else documented otherwise very loudly at the top of the function declaration and/or definition.

What I think I see happening in this PR is that (a) some uses of `idris2_extractInt` are being swapped for explicit representation casts (mostly if not entirely `idris2_vp_to_Int64`) and (b) the default unboxed behavior of `idris2_extractInt` is to assume a 32-bit Int is used. I worry that at the very least this implementation only works when used against the FFI `Int` type because 64-bit integers cannot be unwrapped under the current implementation.

Reply:
Tests were added in the last commit.

8-bit, 16-bit and 32-bit Int are cast to 64-bit at first, And then transformed into pointers. So idris2_vp_to_Int32/Bits32 is almost equivalent to casting the original value 8bit -> 64bit -> 32bit in that order.
I was worried that I was cutting corners with the sign bit handling, but from the test results it seems to be working fine.
(Except for two problems with Int64 that I found accidentally.)

Note: extractInt is not safe at all because it implicitly casts large values and converts them to signed values. Also, now extractInt is only used in ConstCase, so the extract function is not necessary for us if we have a comparison function.

Note: I  agree that documentation should be provided. I will put it on my todo list.